components/authz-service/projectassignment/errors.go

```diff
@@ -0,0 +1,45 @@
+package projectassignment
+
+import (
+	"fmt"
+)
+
+// ProjectsMissingErr occurs when some of the projects in the project diff
+// did not exist.
+type ProjectsMissingErr struct {
+	projectsMissing []string
+}
+
+func NewProjectsMissingError(projectsMissing []string) error {
+	return &ProjectsMissingErr{projectsMissing: projectsMissing}
+}
+
+func (e *ProjectsMissingErr) Error() string {
+	var errorStr string
+	if len(e.projectsMissing) > 1 {
+		errorStr = fmt.Sprintf("You attempted to modify projects for this object but some did not exist: %q", e.projectsMissing)
+	} else {
+		errorStr = fmt.Sprintf("You attempted to modify a project for this object but it did not exist: %q", e.projectsMissing)
+	}
+	return errorStr
+}
+
+// ProjectsUnauthorizedForAssignmentErr occurs when some of the projects in the project diff
+// were not authorized for a specific set of subjects.
+type ProjectsUnauthorizedForAssignmentErr struct {
+	projectsUnauthorized []string
+}
+
+func NewProjectsUnauthorizedForAssignmentError(projectsUnauthorized []string) error {
+	return &ProjectsUnauthorizedForAssignmentErr{projectsUnauthorized: projectsUnauthorized}
+}
+
+func (e *ProjectsUnauthorizedForAssignmentErr) Error() string {
+	var errorStr string
+	if len(e.projectsUnauthorized) > 1 {
+		errorStr = fmt.Sprintf("You attempted to modify projects for this object but you are missing iam:projects:assign on some: %q", e.projectsUnauthorized)
```

Comment:
🤔 Should this be explicit, as you have done, or should it be "You do not have permission to modify these projects: %q" ? Not sure...

Reply:
Not sure either but I think we landed with it being explicit with @susanev. We can have her review all of these error messages when she returns!